xcell/cls/data.py

```diff
@@ -88,24 +229,25 @@ def read_data(self, data_path):
         return data
 
     def get_tracers_used(self, wsp=False):
-        tracers = []
-        for trk, trv in self.data['cls'].items():
-            tr1, tr2 = trk.split('-')
-            if trv['compute'] != 'None':
-                tracers.append(tr1)
-                tracers.append(tr2)
+        lab = ['no_wsp', 'wsp'][wsp]
+        if self.tracers[lab] is None:
+            tr_names = self._get_tracers_defined()
+            tmat = self.get_tracer_matrix()
 
-        tracers_for_cl = []
-        for tr in self.data['tracers'].keys():
-            # Remove tracer number
-            tr_nn = self.get_tracer_bare_name(tr)
-            if tr_nn in tracers:
-                tracers_for_cl.append(tr)
+            # Saving only tr1 to keep order
+            tracers = []
+            for tr1 in tr_names:
+                for tr2 in tr_names:
+                    req = tmat[(tr1, tr2)]
+                    if req['compute'] and (tr1 not in tracers):
+                        tracers.append(tr1)
+                        break
+            if wsp:
+                tracers = self._filter_tracers_wsp(tracers)
```

Comment:
comment: we only keep those that have different masks

Reply:
Done